### ISSUE TITLE
PIP installation of matplotlib latest version fails

### ISSUE THREAD

**Issue by AbinayaSandhiyaM** [matplotlib/matplotlib#21355]
### Bug summary

Hi,

Why been facing below pip installation issue for some modules

Module installation with pip3 starting
ERROR: Could not find a version that satisfies the requirement matplotlib==3.4.1 (from versions: 0.86, 0.86.1, 0.86.2
, 0.91.0, 0.91.1, 1.0.1, 1.1.0, 1.1.1, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.4.0, 1.4.1rc1, 1.4.1, 1.4.2, 1.4.3, 1.5.0, 1.5.1
, 1.5.2, 1.5.3, 2.0.0b1, 2.0.0b2, 2.0.0b3, 2.0.0b4, 2.0.0rc1, 2.0.0rc2, 2.0.0, 2.0.1, 2.0.2, 2.1.0rc1, 2.1.0, 2.1.1,
2.1.2, 2.2.0rc1, 2.2.0, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 3.0.0rc2, 3.0.0, 3.0.1, 3.0.2, 3.0.3, 3.1.0rc1, 3.1.0rc2, 3.1.0,
3.1.1, 3.1.2, 3.1.3, 3.2.0rc1, 3.2.0rc3, 3.2.0, 3.2.1, 3.2.2, 3.3.0rc1, 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.3.4)
ERROR: No matching distribution found for matplotlib==3.4.1

i am able to install matplotlib versions other than 3.4.1.

but it seems available in PyPI site.

### Code for reproduction

```python
pip3 install matplotlib==3.4.1
```

### Actual outcome

Module installation with pip3 starting
ERROR: Could not find a version that satisfies the requirement matplotlib==3.4.1 (from versions: 0.86, 0.86.1, 0.86.2
, 0.91.0, 0.91.1, 1.0.1, 1.1.0, 1.1.1, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.4.0, 1.4.1rc1, 1.4.1, 1.4.2, 1.4.3, 1.5.0, 1.5.1
, 1.5.2, 1.5.3, 2.0.0b1, 2.0.0b2, 2.0.0b3, 2.0.0b4, 2.0.0rc1, 2.0.0rc2, 2.0.0, 2.0.1, 2.0.2, 2.1.0rc1, 2.1.0, 2.1.1,
2.1.2, 2.2.0rc1, 2.2.0, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 3.0.0rc2, 3.0.0, 3.0.1, 3.0.2, 3.0.3, 3.1.0rc1, 3.1.0rc2, 3.1.0,
3.1.1, 3.1.2, 3.1.3, 3.2.0rc1, 3.2.0rc3, 3.2.0, 3.2.1, 3.2.2, 3.3.0rc1, 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.3.4)
ERROR: No matching distribution found for matplotlib==3.4.1

### Expected outcome

it should be installed on RHEL machine

### Operating system

linux

### Matplotlib Version

3.4.1

**Comment by QuLogic** [comment]
> ### Python version
> 
> _No response_

What Python version?

**Comment by tacaswell** [comment]
Are you on a corporate network with an internal pypi mirror?  It is very odd to me that nothing from the 3.4 series.  What do you get for

```
✔ 19:34:15 $ pip3 index versions matplotlib
WARNING: pip index is currently an experimental command. It may be removed/changed in a future release without prior warning.
matplotlib (3.4.3)
Available versions: 3.4.3, 3.4.2, 3.4.1, 3.4.0, 3.3.4, 3.3.3, 3.3.2, 3.3.1, 3.3.0, 3.2.2, 3.2.1, 3.2.0, 3.1.3, 3.1.2, 3.1.1, 3.1.0, 3.0.3, 3.0.2, 3.0.1, 3.0.0, 2.2.5, 2.2.4, 2.2.3, 2.2.2, 2.2.0, 2.1.2, 2.1.1, 2.1.0, 2.0.2, 2.0.1, 2.0.0, 1.5.3, 1.5.2, 1.5.1, 1.5.0, 1.4.3, 1.4.2, 1.4.1, 1.4.0, 1.3.1, 1.3.0, 1.2.1, 1.2.0, 1.1.1, 1.1.0, 1.0.1, 0.91.1, 0.91.0, 0.86.2, 0.86.1, 0.86
  INSTALLED: 3.4.3
  LATEST:    3.4.3

```

**Comment by AbinayaSandhiyaM** [comment]
Hi,

Thanks for the information

We been using RHEL 7.9 machine where python version is 3.6.8

Tried to install matplotlib version 3.4.3 via pip3 installation and faced below issue that pip3 is not able to install matplotlib 3.4 series releases

[root@lirpa-test-abinaya ~]$ pip3 install matplotlib==3.4.3
ERROR: Could not find a version that satisfies the requirement matplotlib==3.4.3 (from versions: 0.86, 0.86.1, 0.86.2, 0.91.0, 0.91.1, 1.0.1, 1.1.0, 1.1.1, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.4.0, 1.4.1rc1, 1.4.1, 1.4.2, 1.4.3, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 2.0.0b1, 2.0.0b2, 2.0.0b3, 2.0.0b4, 2.0.0rc1, 2.0.0rc2, 2.0.0, 2.0.1, 2.0.2, 2.1.0rc1, 2.1.0, 2.1.1, 2.1.2, 2.2.0rc1, 2.2.0, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 3.0.0rc2, 3.0.0, 3.0.1, 3.0.2, 3.0.3, 3.1.0rc1, 3.1.0rc2, 3.1.0, 3.1.1, 3.1.2, 3.1.3, 3.2.0rc1, 3.2.0rc3, 3.2.0, 3.2.1, 3.2.2, 3.3.0rc1, 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.3.4)
ERROR: No matching distribution found for matplotlib==3.4.3

But able to install via 3.3.4 without any issue with python 3.6

Now clearly understand that latest version of python would require to install latest version of matplotlib.

thanks for your time and have a great day!

**Comment by Snow1703** [comment]
Hi,

Thanks for the information 
 I am trying to install in python IDLE SHELL 3.13.2 , when i go to install mathplotlib , I am getting 
ERROR like : Collecting matplotlib==3.4.1
  Using cached matplotlib-3.4.1.tar.gz (37.3 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [42 lines of output]
      <string>:34: SetuptoolsDeprecationWarning: The test command is disabled and references to it are deprecated.
      !!
      
              ********************************************************************************
              Please remove any references to `setuptools.command.test` in all supported versions of the affected package.
      
              This deprecation is overdue, please update your project and remove deprecated
              calls to avoid build errors in the future.
              ********************************************************************************
      
      !!
      Traceback (most recent call last):
        File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 389, in <module>
          main()
          ~~~~^^
        File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 373, in main
          json_out["return_val"] = hook(**hook_input["kwargs"])
                                   ~~~~^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 143, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/private/var/folders/cr/yyc9tg0x3_dbhqjtqfbpj_9c0000gn/T/pip-build-env-x047xr4v/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 334, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/cr/yyc9tg0x3_dbhqjtqfbpj_9c0000gn/T/pip-build-env-x047xr4v/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 304, in _get_build_requires
          self.run_setup()
          ~~~~~~~~~~~~~~^^
        File "/private/var/folders/cr/yyc9tg0x3_dbhqjtqfbpj_9c0000gn/T/pip-build-env-x047xr4v/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 522, in run_setup
          super().run_setup(setup_script=setup_script)
          ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/cr/yyc9tg0x3_dbhqjtqfbpj_9c0000gn/T/pip-build-env-x047xr4v/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
          ~~~~^^^^^^^^^^^^^^^^
        File "<string>", line 54, in <module>
        File "/private/var/folders/cr/yyc9tg0x3_dbhqjtqfbpj_9c0000gn/T/pip-install-qd00jmh6/matplotlib_73d3a54869124df3bc3d6058f59ba3e6/versioneer.py", line 1410, in get_version
          return get_versions()["version"]
                 ~~~~~~~~~~~~^^
        File "/private/var/folders/cr/yyc9tg0x3_dbhqjtqfbpj_9c0000gn/T/pip-install-qd00jmh6/matplotlib_73d3a54869124df3bc3d6058f59ba3e6/versioneer.py", line 1344, in get_versions
          cfg = get_config_from_root(root)
        File "/private/var/folders/cr/yyc9tg0x3_dbhqjtqfbpj_9c0000gn/T/pip-install-qd00jmh6/matplotlib_73d3a54869124df3bc3d6058f59ba3e6/versioneer.py", line 401, in get_config_from_root
          parser = configparser.SafeConfigParser()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by QuLogic** [comment]
Matplotlib 3.4.1 doesn't support Python 3.13; it was released 3.5 years before.

**Comment by junior8542** [comment]
Note: you may need to restart the kernel to use updated packages.
ERROR: Could not find a version that satisfies the requirement mathplotlib (from versions: none)
ERROR: No matching distribution found for mathplotlib
bonjour j'ai ce problème que je n'arrive pas à résoudre
